src/leappto/actor_cli.py

```diff
@@ -0,0 +1,265 @@
+        dest="excluded_paths",
+        nargs='*',
+        type=_path_spec,
+        help='define paths which will be excluded from the source'
+    )
+    migrate_cmd.add_argument("--ignore-default-port-map", default=False,
+                             help='Default port mapping detected by leapp toll will be ignored', action="store_true")
+    migrate_cmd.add_argument('--container-name', '-n', default=None,
+                             help='Name of new container created on target host')
+    migrate_cmd.add_argument(
+        '--force-create',
+        action='store_true',
+        help='force creation of new target container, even if one already exists'
+    )
+    migrate_cmd.add_argument('--disable-start', dest='disable_start', default=False,
+                             help='Migrated container will not be started immediately', action="store_true")
+    migrate_cmd.add_argument('--target-user', default="root", help='Connect as this user to the target via ssh')
+    migrate_cmd.add_argument('--source-user', default="root", help='Connect as this user to the source via ssh')
+    migrate_cmd.add_argument('--debug', default=False, action="store_true", help='Turn on debug logging on stderr')
+
+
+def _migrate_machine(arguments):
+    default_excluded_paths = ['/dev/*', '/proc/*', '/sys/*', '/tmp/*', '/run/*', '/mnt/*', '/media/*', '/lost+found/*']
+    data = {
+        "target_host": _make_base_object(arguments.target),
+        "source_host": _make_base_object(arguments.machine),
+        "tcp_ports_user_mapping": _to_port_map(arguments.forwarded_tcp_ports or ()),
+        "excluded_tcp_ports": {"tcp": {str(x[0]): {"name": ""} for x in arguments.excluded_tcp_ports or ()}},
+        "excluded_paths": {"value": arguments.excluded_paths or default_excluded_paths},
+        "start_container": _make_base_object(not arguments.disable_start),
+        "target_user_name": _make_base_object(arguments.target_user),
+        "source_user_name": _make_base_object(arguments.source_user),
+        "force_create": _make_base_object(arguments.force_create),
+        "user_container_name": _make_base_object(arguments.container_name or ''),
+    }
+    if arguments.debug or not arguments.print_port_map:
+        logging.getLogger().addHandler(logging.StreamHandler(sys.stderr))
+    return data, 'migrate-machine' if not arguments.print_port_map else 'port-mapping'
+
+
+@contextmanager
+def _stdout_socket():
+    directory = tempfile.mkdtemp('', 'LEAPP_STDOUT', None)
+    name = os.path.join(directory, 'leapp_stdout.sock')
+    registry.register_environment_variable('LEAPP_ACTOR_STDOUT_SOCK', name)
+    # This might be wanted to be a bit more dynamic but well it's good enough for now
+    registry.register_environment_variable('LEAPP_ACTOR_OUTPUT', '/usr/bin/actor-stdout')
+
+    env = os.environ.copy()
+    env["LEAPP_ACTOR_STDOUT_SOCK"] = name
+    p = subprocess.Popen(["actor-stdout", "server"], env=env)
+
+    yield
+    if p.poll():
```

Comment:
shouldn't  we have a try caluse here rather than that if?

https://docs.python.org/2/library/contextlib.html

At the point where the generator yields, the block nested in the with statement is executed. The generator is then resumed after the block is exited. If an unhandled exception occurs in the block, it is reraised inside the generator at the point where the yield occurred. Thus, you can use a try…except…finally statement to trap the error (if any), or ensure that some cleanup takes place. If an exception is trapped merely in order to log it or to perform some action (rather than to suppress it entirely), the generator must reraise that exception. Otherwise the generator context manager will indicate to the with statement that the exception has been handled, and execution will resume with the statement immediately following the with statement.

Reply:
good point